### PR TITLE
feat(console): show SSO identities on user details page and reorg the page

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

1. reorg the user details page and make SSO identities visible.
2. update `GET /api/users/:userId` and corresponding zod guard/type definitions to return the user's (characterized using `userId`) SSO identities.
 - Update SSO identity queries as well since the previous syntax is not right.

## Testing

Tested locally.
image
image

## Checklist

~~- [ ] `.changeset`~~
~~- [ ] unit tests~~
~~- [x] integration tests~~
~~- [ ] necessary TSDoc comments~~

### CODE REVIEW

File: packages/console/src/pages/UserDetails/UserSettings/components/UserSsoIdentities/index.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          data.find((ssoConnector) => ssoConnector.id === identity.ssoConnectorId) ?? {};
+        const { identityId: userIdentity, issuer } = identity;
+
+        if (!(logo && name)) {
+          return;
+        }
+
+        return { logo, name, userIdentity, issuer };
+      })
+      .filter(Boolean) as DisplayConnector[];
```
Comment (wangsijie) [inline packages/console/src/pages/UserDetails/UserSettings/components/UserSsoIdentities/index.tsx:53]: nit: `.filter((identity): identity is DisplayConnector => Boolean(identity))`

Comment (darcyYe) [reply packages/console/src/pages/UserDetails/UserSettings/components/UserSsoIdentities/index.tsx]: Updated

File: packages/core/src/queries/user-sso-identities.ts [hunk 2]
```diff
@@ -24,9 +25,19 @@ export default class UserSsoIdentityQueries extends SchemaQueries<
   ): Promise<Nullable<UserSsoIdentity>> {
     return this.pool.maybeOne<UserSsoIdentity>(sql`
       select *
-      from ${UserSsoIdentities.table}
-      where ${UserSsoIdentities.fields.issuer} = ${issuer}
-      and  ${UserSsoIdentities.fields.identityId} = ${ssoIdentityId}
+      from ${sql.identifier([UserSsoIdentities.table])}
+      where ${sql.identifier([UserSsoIdentities.fields.issuer])} = ${issuer}
+      and ${sql.identifier([UserSsoIdentities.fields.identityId])} = ${ssoIdentityId}
```
Comment (wangsijie) [inline packages/core/src/queries/user-sso-identities.ts:30]: Is it necessary to wrap `identifier`?

Comment (darcyYe) [reply packages/core/src/queries/user-sso-identities.ts:30]: yes, added the new query method and API integration tests. The syntax error was found by IT test cases and we wrapped the identifier to address this syntax error.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**